types/mysql__xdevapi/index.d.ts

```diff
@@ -0,0 +1,493 @@
+// Type definitions for @mysql/xdevapi 8.0
+// Project: https://github.com/mysql/mysql-connector-nodejs
+// Definitions by: Andrew Rioux <https://github.com/r-a303931>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.3
+
+export type DataModel = number;
+export type DocumentOrJSON = any;
+export type ParserOptions = DataModel;
+export type SearchConditionStr = string;
+
+export enum LockContention {
+	DEFAULT = 0,
+	NOWAIT = 1,
+	SKIP_LOCKED = 2
+}
+
+export interface CollectionAddOptions {
+	upsert: boolean;
+}
+
+export interface CreateCollectionOptions {
+	ReuseExistingObject?: boolean;
+}
+
+export interface FieldDefinition {
+	field: string;
+	type: string;
+	required?: boolean;
+	options?: number;
+	srid?: number;
+}
+
+export interface IndexDefinition {
+	type?: 'INDEX' | 'SPATIAL';
+	fields: FieldDefinition[];
+}
+
+export interface SocketFactory {
+	// tslint:disable-next-line
+	createSocket: Function;
```

Comment:
As discussed [here](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes), it's not good practice to use `Function`.